pravega/pravega
# Issue 2547: Ensure ClientFactory.close() always closes the Controller client.

**Change log description**
- Ensure `ClientFactory.close()` always closes the controller client.
- Ensure `StreamMetadataResourceImpl` closes `ClientFactory` which is created while processing `getReaderGroup()` API.

**Purpose of the change**
Fixes #2547 

**What the code does**
- Ensure `ClientFactory.close()` always closes the controller client, irrespective of whether the controllerclient is created internally or passed to it.
- Ensure `StreamMetadataResourceImpl` closes `ClientFactory` which is created while processing `getReaderGroup()` API.

**How to verify it**
All the existing tests should pass. Verified that the open sockets are closed after invocation of close() using netstat command.
